subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/CompilationWithAnnotationProcessingResult.java

```diff
@@ -19,12 +19,17 @@
 import org.gradle.api.internal.tasks.compile.incremental.processing.AnnotationProcessingResult;
 import org.gradle.workers.internal.DefaultWorkResult;
 
-public class JdkJavaCompilerResult extends DefaultWorkResult {
+public class CompilationWithAnnotationProcessingResult extends DefaultWorkResult {
```

Comment:
The name was deliberately not related to annotation processing, as this result is always used, no matter if we process or not. It's the result we get when using the JDK API compiler (as opposed to invoking the `javac` command where we get no insights).